documentation/test_python/pybind_signatures/pybind_signatures.cpp

```diff
@@ -2,6 +2,7 @@
 #include <pybind11/pybind11.h>
 #include <pybind11/stl.h> /* needed for std::vector! */
 #include <pybind11/functional.h> /* for std::function */
+#include <pybind11/stl/filesystem.h> /* for std::filesystem::path */
```

Comment:
The `std::filesystem` casters are implemented in pybind11 2.7 (https://github.com/pybind/pybind11/commit/5bcaaa0423c6757ca1c2738d0a54947dacdb03a1 and then moved to a dedicated header in https://github.com/pybind/pybind11/commit/6d1b197b4669c70da94c8efb516374e1935d3ea3). 

You're lucky that this test already contains opt-in parts for pybind11 2.3 and 2.6, in the form of a dedicated `MyClass23` / `MyClass26`: https://github.com/mosra/m.css/blob/c5e9b803229ce15a103daf9f4b7612a244170a52/documentation/test_python/pybind_signatures/pybind_signatures.cpp#L118-L153
So if you want to tackle addition of a version-dependent test yourself, it'd be about putting the new function into a `MyClass27` (hiding them, and the include, in an appropriate version `#ifdef`) and then doing something similar as in https://github.com/mosra/m.css/blob/c5e9b803229ce15a103daf9f4b7612a244170a52/documentation/test_python/test_pybind.py#L424-L427 to verify the output, and same for the stub verification right below.